webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,247 @@
+      <Button
+        type="primary"
+        onClick={() => {
+          form.validateFields().then(values => {
+            handleCreateEnvironment(values);
+          }).catch();
+        }
+        }>Launch</Button>
+    </>}
+    title="Launch JupyterLab"
+    visible={visible}
+    {...props}>
+    {showFullConfig?
+      <Form form={form}>
+        <div className={css.note}
+        >
+          <Link external path="/docs/reference/command-notebook-config.html">
+          Read about notebook settings
+          </Link>
+        </div><React.Suspense
+          fallback={<div className={css.loading}><Spinner className="minHeight" /></div>}>
+          <Item
+            name="config"
+            rules={[ { message: 'Invalid YAML', required: true }, () => ({
+              validator(_, value) {
+                try {
+                  yaml.load(value);
+                  return Promise.resolve();
+                } catch(err) {
+                  return Promise.reject(new Error('Invalid YAML'));
+                }
+              },
+            }) ]}>
+            <MonacoEditor
+              height={400}
+              language="yaml"
+              options={{
+                minimap: { enabled: false },
+                scrollBeyondLastLine: false,
+                selectOnLineNumbers: true,
+              }} />
+          </Item>
+        </React.Suspense>
+      </Form> :
+      <Form
+        form={form}
+        initialValues={storage.getWithDefault(STORAGE_KEY, {
+          slots:1,
+          template: 'default',
+          type: undefined,
+        })}
+        labelCol={{ span:8 }}
+        onValuesChange={storeConfig}>
+        <Item label="Notebook Template" name="template">
+          <Select>
+            <Option key="default" value="default">Default Task Template</Option>
+            {templates.map(temp =>
+              <Option key={temp.name} value={temp.name}>{temp.name}</Option>)}
+          </Select>
+        </Item>
+        <Item label="Name" name="name">
+          <Input placeholder="Name" />
+        </Item>
+        <Item
+          label="Resource Pool"
+          name="pool"
+          rules={[ { message: 'Select a resource pool', required: true } ]}>
+          <Select
+            placeholder="Select a resource pool"
+            onChange={handleResourcePoolUpdate}>
+            {resourcePools.map(pool =>
+              <Option key={pool.name} value={pool.name}>{pool.name}</Option>)}
+          </Select>
+        </Item>
+        <Row justify="end">
+          <Col span={8}>
+            <Item
+              label="Type"
+              labelCol={{ span:8 }}
+              name="type"
+              rules={[ { message: 'Select a resource type', required: true } ]}>
+              <RadioGroup
+                options={resourceTypeOptions}
+                onChange={handleTypeUpdate} />
+            </Item>
+          </Col>
```

Comment:
**style:** minor adjustment for mobile view

![2021-05-20 at 10 53 AM](https://user-images.githubusercontent.com/220971/119038213-16819980-b970-11eb-8357-834d99d42587.png)

this looks ok when both options are present but looks off when it's only `Type` (no Number of Slots field)